pkg/controller/infinispan/util/kubernetes.go

```diff
@@ -199,6 +199,26 @@ func (k Kubernetes) PublicIP() string {
 	return u.Hostname()
 }
 
+// GetNodesHost return the addresses of the k8s nodes
+func (k Kubernetes) GetNodesHost() []string {
+	nodes := &v1.NodeList{}
+	k.Client.List(context.TODO(), nodes)
+	nodeHosts := make([]string, nodes.Size())
+	for i, n := range nodes.Items {
+		for _, v := range n.Status.Addresses {
+			if v.Type == v1.NodeHostName {
+				// Fallback using hostname if internal IP address is not available
+				nodeHosts[i] = v.Address
+			}
+			if v.Type == v1.NodeInternalIP {
+				nodeHosts[i] = v.Address
+				break
+			}
+		}
+	}
+	return nodeHosts
```

Comment:
@rigazilla looks like we need to trunc slice to prevent empty values or choose another solution

Reply:
I see what you mean, let me clean up

Reply:
@dmvolod I've changed the impl. If it's ok I'll port also on #416

Reply:
Thanks, @rigazilla, looks better